packages/components/src/components/editor/condition-builder/conditionGroup.tsx

```diff
@@ -0,0 +1,53 @@
+import { IvyIcons } from '@axonivy/ui-icons';
+import { BasicSelect } from '@/components/common/select/select';
+import { Flex } from '@/components/common/flex/flex';
+import { Label } from '@radix-ui/react-label';
+import { Button } from '@/components/common/button/button';
+import { cn } from '@/utils/class-name';
+import { conditionBorder } from '@/components/editor/condition-builder/conditionBuilder.css';
+import { Condition, type ConditionProps } from './condition';
+import {
+  useConditionBuilderContext,
+  type ConditionGroupData,
+  type LogicOperator
+} from '@/components/editor/condition-builder/conditionBuilderContext';
+
+interface ConditionGroupProps extends Pick<ConditionProps, 'groupIndex'> {
+  group: ConditionGroupData;
+  groupCount: number;
+}
+
+export const ConditionGroup = ({ group, groupIndex, groupCount }: ConditionGroupProps) => {
+  const { updateLogicalOperator, addCondition, removeConditionGroup, conditionMode, logicalOperatorOptions } = useConditionBuilderContext();
+  return (
+    <Flex direction='column' gap={2} className='ui-condition-builder-group' data-condition-mode={conditionMode}>
+      <Flex direction='column' className={conditionMode === 'nested-condition' ? cn(conditionBorder) : ''} gap={2}>
```

Comment:
didn't you check what `cn` does?
```suggestion
      <Flex direction='column' className={cn(conditionBorder, conditionMode === 'nested-condition' && 'nested-condition')} gap={2}>
```

or why not add this mode as data arg? sounds more like a data arg than a classname...
```suggestion
      <Flex direction='column' className={conditionBorder} data-condition-mode={conditionMode} gap={2}>
```